osquery/tables/system/tests/darwin/system_extensions_test.cpp

```diff
@@ -0,0 +1,96 @@
+/**
+ * Copyright (c) 2014-present, The osquery authors
+ *
+ * This source code is licensed as defined by the LICENSE file found in the
+ * root directory of this source tree.
+ *
+ * SPDX-License-Identifier: (Apache-2.0 OR GPL-2.0-only)
+ */
+
+#include <boost/algorithm/string.hpp>
+#include <boost/property_tree/json_parser.hpp>
+#include <boost/property_tree/ptree.hpp>
+#include <gtest/gtest.h>
+
+#include <osquery/config/tests/test_utils.h>
+#include <osquery/core/sql/query_data.h>
+#include <osquery/logger/logger.h>
+
+namespace pt = boost::property_tree;
+
+namespace osquery {
+namespace tables {
+
+extern QueryData genExtensionsFromPtree(pt::ptree& ptree);
+
+class SystemExtensionTests : public testing::Test {};
+
+TEST_F(SystemExtensionTests, test_parse_plist) {
+  pt::ptree ptree;
+  std::stringstream ss;
+  ss << "{ \"version\": \"1\", \"extensions\": [{ \"state\": "
```

Comment:
Some of the other tests store the mock file as a file in the test config directory.
E.g., `auto dbPlist_path = getTestConfigDirectory() / "db.plist";`

Which currently is here: https://github.com/osquery/osquery/tree/master/tools/tests/configs